core/controllers/topics_and_skills_dashboard.py

```diff
@@ -69,14 +69,12 @@ class NewSkillHandler(base.BaseHandler):
     def post(self):
         if not feconf.ENABLE_NEW_STRUCTURES:
             raise self.PageNotFoundException
-        topic_id = None
         topic_id = self.payload.get('topic_id')
 
         if topic_id is not None:
             topic = topic_services.get_topic_by_id(topic_id, strict=False)
             if topic is None:
-                raise self.PageNotFoundException(
-                    Exception('The topic with the given id doesn\'t exist.'))
+                raise self.InvalidInputError
```

Comment:
Did you check the name of this, or test this case? (My bad though, I think I gave you the wrong name.)

In general we should aim to have tests that (at least) cover every line of code changed.

Reply:
Oh, yeah, its InvalidInputException. Done, also, added a test that checks this case.